### PR TITLE
Feature/#5045 more generics in RxJavaPlugins

### PR DESCRIPTION
**Pull request by jschneider**
I widened the bounds for the functions. Looks good for me.
Related to #5045

### CODE REVIEW

File: src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java [hunk 1]
```diff
@@ -16,14 +16,26 @@
 
 package io.reactivex.plugins;
 
-import io.reactivex.*;
+import io.reactivex.Completable;
```
Comment (akarnokd) [inline src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java]: Please keep imports as *

**Review by akarnokd — CHANGES_REQUESTED** [review]

**Comment by akarnokd** [comment]
Looks like this needs rebasing. I'm sorrry, I'm trying to merge your PRs in proper order but multiple recent PRs change the same file(s).

**Comment by jschneider** [comment]
Yeah, no problem. Rebasing isn't too difficult ;-)

File: src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java (hunk 1 above)
Comment (jschneider) [reply src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java]: Yes. Changed the code style settings accordingly.
Do you have a code style configuration for IntelliJ Idea?

**Comment by jschneider** [comment]
fixed import and rebased

File: src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java [hunk 2]
```diff
@@ -17,13 +17,14 @@
 package io.reactivex.plugins;
 
 import io.reactivex.*;
-import io.reactivex.Observable;
-import io.reactivex.Observer;
-import io.reactivex.Scheduler.Worker;
-import io.reactivex.disposables.*;
+import io.reactivex.disposables.Disposable;
```
Comment (akarnokd) [inline src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java]: Could you also restore these import changes to their * form?

**Comment by akarnokd** [comment]
Looks like there is a [compilation error](https://travis-ci.org/ReactiveX/RxJava/builds/198621003#L190) with the changes.

**Comment by jschneider** [comment]
Uups. Fixed it (at least I hope)

File: src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java [hunk 3]
```diff
@@ -17,13 +17,15 @@
...
-import io.reactivex.Observable;
-import io.reactivex.Observer;
 import io.reactivex.Scheduler.Worker;
-import io.reactivex.disposables.*;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.disposables.Disposables;
 import io.reactivex.exceptions.TestException;
 import io.reactivex.flowables.ConnectableFlowable;
-import io.reactivex.functions.*;
+import io.reactivex.functions.BiFunction;
```
Comment (akarnokd) [inline src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java:25]: Please restore these as well.

**Review by akarnokd — APPROVED** [review]

**Pull request merged**